pkgs/tools/system/incron/default_path.patch

```diff
@@ -0,0 +1,36 @@
+diff --git usertable.cpp usertable.cpp
+index 11fd04b..a8681bd 100644
+--- a/usertable.cpp
++++ b/usertable.cpp
+@@ -43,9 +43,6 @@
+ #define DONT_FOLLOW(mask) (false)
+ #endif // IN_DONT_FOLLOW
+ 
+-// this is not enough, but...
+-#define DEFAULT_PATH "/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin"
+-
+ 
+ PROC_MAP UserTable::s_procMap;
+ 
+@@ -597,12 +594,20 @@ void UserTable::RunAsUser(std::string cmd) const
+     if (clearenv() != 0)
+       goto failed;
+ 
++    // try to recreate the user path as best as possible
```

Comment:
The profiles are arbitrary. `NIX_PROFILES` environment variable would contain all of them but that probably will not be available to the systemd service. Instead, it is common to provide `extraPackages` option in the NixOS module and then add them to the service ([example](https://github.com/NixOS/nixpkgs/blob/9b2f1145e7fecea81e7be7a0a2dbaee20effc8bc/nixos/modules/services/networking/firewall.nix#L554)).

Reply:
How would you recommend doing this on a per user basis? Or are you just suggesting to skip that entirely and have the extraPackages cover all users? If so, would that not restrict users somewhat as they can't modify configuration.nix?

Reply:
Well would not be able to install packages to global profiles either, would they?

Reply:
No, but that is why ~/.nix-profile/bin is included in the path as well.

Reply:
Oh right, `pwd->pw_name` is probably the user’s name. Maybe this would be good enough until the upstream adds user service support, if a comment that this is does not support other profiles than the common ones is added.

Reply:
I've added a comment... please suggest any changes to wording.

Reply:
I've modified this so the user can explicitly set the path for the system incrontab. Each users incrontab will still rely on the packages available to that user, but that can still work out reasonably well considering the users.users.<name>.packages option.

Reply:
On the other hand, the original code also does not try to emulate user’s PATH, so maybe we can just take the `config.system.path` from the service.